docs/topics/async.txt

```diff
@@ -63,81 +63,88 @@ deploy Django using :doc:`ASGI </howto/deployment/asgi/index>` instead.
+    results = sync_to_async(Blog.objects.get)(pk=123)
 
 You may find it easier to move any ORM code into its own function and call that
-entire function using :func:`sync_to_async`. If you accidentally try to call
-part of Django that is still synchronous-only from an async view, you will
-trigger Django's :ref:`asynchronous safety protection <async-safety>` to
-protect your data from corruption.
+entire function using :func:`sync_to_async`. For example::
+
+    from asgiref.sync import sync_to_async
+
+    @sync_to_async
+    def get_blog(pk):
+        return Blog.objects.select_related('author').get(pk=pk)
+
+If you accidentally try to call part of Django that is still synchronous-only
+from an async view, you will trigger Django's
+:ref:`asynchronous safety protection <async-safety>` to protect your data from
+corruption.
 
 Performance
 -----------
 
-When running in a mode that does not match the view (e.g. an async view under
+When running in a mode that does not match the view (i.e. an async view under
 WSGI, or a traditional sync view under ASGI), Django must emulate the other
 call style to allow your code to run. This context-switch causes a small
 performance penalty of around a millisecond.
 
-This is true of middleware as well, however. Django will attempt to minimize
-the number of context-switches. If you have an ASGI server, but all your
-middleware and views are synchronous, it will switch just once, before it
-enters the middleware stack.
+This is also true of middleware as well. Django will attempt to minimize the
+number of context-switches. If you have an ASGI server, but all your middleware
+and views are synchronous, it will switch just once, before it enters the
+middleware stack.
 
-If, however, you put synchronous middleware between an ASGI server and an
+However, if you put synchronous middleware between an ASGI server and an
 asynchronous view, it will have to switch into sync mode for the middleware and
-then back to asynchronous mode for the view, holding the synchronous thread
-open for middleware exception propagation. This may not be noticeable, but bear
-in mind that even adding a single piece of synchronous middleware can drag your
-whole async project down to running with one thread per request, and the
-associated performance penalties.
-
-You should do your own performance testing to see what effect ASGI vs. WSGI has
-on your code. In some cases, there may be a performance increase even for
+then back to async mode for the view. Django will also hold the sync thread
+open for middleware exception propagation. This may not be noticeable at first,
+but adding this penalty of one thread per request can remove any async
+performance advantage.
+
+You should do your own performance testing to see what effect ASGI versus WSGI
+has on your code. In some cases, there may be a performance increase even for
 purely-synchronous codebase under ASGI because the request-handling code is
```

Comment:
I think we need an "a" before "purely-synchronous"

Reply:
Yes. Also dropping this unnecessary-hyphen.